src/devices/oregon_scientific.c

```diff
@@ -806,16 +811,46 @@ static int oregon_scientific_v3_decode(r_device *decoder, bitbuffer_t *bitbuffer
         return 1;
     }
     else if ((msg[0] == 0x20) || (msg[0] == 0x21) || (msg[0] == 0x22) || (msg[0] == 0x23) || (msg[0] == 0x24)) { // Owl CM160 Readings
-        msg[0] = msg[0] & 0x0f;
+
+        int channel = msg[0]>>4; //let's grab the channel before we manipulate the msg for checksum validation
```

Comment:
But we know `(msg[0] == 0x20) || (msg[0] == 0x21) || (msg[0] == 0x22) || (msg[0] == 0x23) || (msg[0] == 0x24)`
So `msg[0]>>4` must be `== 2`, right?

Reply:
I see, was the original code maybe `swap_nibbles(msg[0]) >> 4`? Then we'd get 0,1,2,3,4 which sounds like a channel, yes.
Use `msg[0] & 0xf` then.

Reply:
Agree and changes accordingly.